### ISSUE TITLE
Can "*.nc" patterns be supported?

### ISSUE THREAD

**Issue by agstephens** [intake/intake-esm#385]
**Is your feature request related to a problem? Please describe.**

I would like to be able to include glob patterns (`"*.nc"`) in the paths of my Intake-esm catalogs. At present, the code uses `xr.open_dataset()` to open netcdf files. I have cases in which the Intake catalog is on a POSIX file system so it is reasonable to provide the data path as `"/foo/baa/*.nc"` and the code could use `xarray` to load a multi-file dataset (using `xr.open_mfdataset()`.

**Describe the solution you'd like**

The solution might be as simple as:

```
diff --git a/intake_esm/source.py b/intake_esm/source.py
index ef5d758..1216e0c 100644
--- a/intake_esm/source.py
+++ b/intake_esm/source.py
@@ -51,7 +51,13 @@ def _open_dataset(
     else:
         url = fsspec.open(urlpath, **storage_options).open()

-    ds = xr.open_dataset(url, **xarray_open_kwargs)
+    # Handle multi-file datasets with `xr.open_mfdataset()`
+    if "*" in url or isinstance(url, list):
+         _open_dataset = xr.open_mfdataset
+    else:
+         _open_dataset = xr.open_dataset
+
+    ds = _open_dataset(url, **xarray_open_kwargs)
```

This is basically how the `intake_xarray` library works:

https://github.com/intake/intake-xarray/blob/master/intake_xarray/netcdf.py#L65-L84

**Describe alternatives you've considered**

There may be clear design principle that would be broken by this proposed change. Please let me know if that is the case, and whether there is another more appropriate way of resolving my problem. :-)

**Comment by andersy005** [comment]
@agstephens, 100% agree it would be useful to support this use case. If you have time, feel free to take a look at the _improved_ `_open_dataset` function on the main branch in https://github.com/intake/intake-esm/blob/aaa71db20bc768d5569ca7bcb1c3c27c6d3499ef/intake_esm/source.py#L34-L45

A PR is welcome :) No pressure though (I'm happy to look into this).

**Comment by agstephens** [comment]
Hi @andersy005, I got the tests running and have created the PR here: #393. Many thanks.

**Comment by andersy005** [comment]
Addressed in #395
